### PR TITLE
Complete makeover

### PR DESCRIPTION
**Pull request by otter18**

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -1,188 +1,100 @@
 #!/usr/bin/env python3
-'''''''''''''''''''''''''''''
```
Comment (lesterrry) [inline src/main.py:2]: Bruh don't be lame

File: src/main.py [hunk 2]
```diff
@@ -1,188 +1,100 @@
...
-import gc
+from telethon import TelegramClient, events
+from telethon.tl.types import PeerUser
 
+from cronitor_logging import Cronitor, cronitor_heartbeat
+from log import log_response, setup_excepthook
+from utils import Phonebook, UserType
 
 CONFIG_FILE_NAME = "config"  # Your .toml config file name (like 'file')
 VERSION = "0.3.0-beta"
```
Comment (lesterrry) [inline src/main.py]: Go to `v1.0.0-beta`

File: src/cronitor_logging.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
+#  Copyright (c) Lester Covey (me@lestercovey.ml) & ChernV (@otter18), 2022
+
+import requests
+import asyncio
+from utils import get_log_string
+
+
+async def cronitor_heartbeat(cronitor_obj):
```
Comment (lesterrry) [inline src/cronitor_logging.py]: Nobody uses `..._obj` in names lol

File: src/log.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        if exc_type is KeyboardInterrupt:
+            sys.__excepthook__(exc_type, value, traceback)
+            return
+
+        err_str = f"FATAL: {value}"
+        print(err_str, file=sys.stderr)
+        logging.exception(err_str)
+        if cronitor is not None:
+            try:
+                cronitor.cronitor_informstate(err_str, "fail")
```
Comment (lesterrry) [inline src/log.py]: What? This name is not even defined

File: src/log.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+            except Exception:
+                pass
+
+        sys.exit(1)
+
+    sys.excepthook = my_except_hook
+
+    @atexit.register
+    def atexit_func():
+        cronitor.informstate(f"Exiting Pong...", "complete")
```
Comment (lesterrry) [inline src/log.py]: Unused f-string

File: src/main.py [hunk 6]
```diff
@@ -1,188 +1,100 @@
...
 log_str = f"Starting Pong v{VERSION}..."
 print(log_str)
+
 if config['service']['logging_enabled']:
-	logging.info(log_str)
+    logging.info(log_str)
+
 if config['service']['cronitor_integrated']:
-	cronitor_informstate(log_str, "run")
-del CONFIG_FILE_NAME, VERSION, setup_mode, log_str
```
Comment (lesterrry) [inline src/main.py:183]: Why removed

File: src/main.py [hunk 7]
```diff
@@ -1,188 +1,100 @@
...
-gc.collect()
+    cronitor.informstate(log_str, "run")
+
 loop = client.start().loop
 if config['service']['cronitor_integrated']:
-	loop.create_task(cronitor_heartbeat())
```
Comment (lesterrry) [inline src/main.py:187]: I want my tabs back

**Review by lesterrry — COMMENTED** [review]
Bruh I need some changes

File: src/cronitor_logging.py [hunk 8]
```diff
@@ -1,7 +1,9 @@
 #  Copyright (c) Lester Covey (me@lestercovey.ml) & ChernV (@otter18), 2022
 
-import requests
 import asyncio
+
+import requests
```
Comment (lesterrry) [inline src/cronitor_logging.py:4]: Why?

Comment (lesterrry) [reply src/cronitor_logging.py:4]: why extra line

**Review by lesterrry — COMMENTED** [review]
Better but previous issues still need addressing

File: src/main.py (hunk 7 above)
Comment (otter18) [reply src/main.py:187]: Fine, you can get them

File: src/main.py (hunk 6 above)
Comment (otter18) [reply src/main.py:183]: Useless

Comment (lesterrry) [reply src/main.py:183]: I mean :183

File: src/log.py (hunk 4 above)
Comment (otter18) [reply src/log.py]: Nope. It is defined in line 19

`def setup_excepthook(config, cronitor):`

File: src/main.py (hunk 6 above)
Comment (otter18) [reply src/main.py:183]: Still useless. Python takes care of memory by itself

File: src/main.py (hunk 1 above)
Comment (otter18) [reply src/main.py:2]: fixed

File: src/main.py (hunk 2 above)
Comment (otter18) [reply src/main.py]: alright

File: src/cronitor_logging.py (hunk 3 above)
Comment (otter18) [reply src/cronitor_logging.py]: _bruh, why not_

although I fixed it

File: src/log.py (hunk 5 above)
Comment (otter18) [reply src/log.py]: fixed

File: src/log.py (hunk 4 above)
Comment (lesterrry) [reply src/log.py]: Still in progress

**Pull request merged**